### PR TITLE
Update all.md

### PR DESCRIPTION
**Pull request by shivangnayar-dev**
added new content (definitions, codeblocks, etc.)

### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by shivangnayar-crypto** [comment]
@SSwiniarski

**Comment by shivangnayar-crypto** [comment]
I need help regarding how to pass those ci checks

**Comment by SSwiniarski** [comment]
@shivangnayar-dev This PR doesn't address the issue you were assigned. The issue was to add codebyte entries to two terms. This only addresses the one term and does not add a codebyte, but just rewrites the existing example code. Do you want to revise this PR to address the issue, or do you want to be unassigned from the issue and have this PR reviewed on its own merits in its current state?

**Comment by shivangnayar-dev** [comment]
i apologies , I will do it can you help me with providing more details like what actually you need.
please assign me this issue.
thanks

**Comment by SSwiniarski** [comment]
@shivangnayar-dev what the issue is asking for is to add new Codebytes at the end of the two entries. Codebytes are examples of code that can be run by the reader of the entry. They should be different from the example given in the body of the entry, and should look something like this:

````
## Codebyte Example

A sentence describing what the codebyte does:

```codebyte/javascript
the example code to run goes here
```
````

You can look at the codebytes for other JavaScript entries, like [here.](https://github.com/Codecademy/docs/edit/main/content/javascript/concepts/arrays/terms/indexOf/indexOf.md)

The issue you claimed wants codebytes to be added to both these entries:
- https://www.codecademy.com/resources/docs/javascript/promise/all
- https://www.codecademy.com/resources/docs/javascript/promise/any

File: content/javascript/concepts/promise/terms/all/all.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 ---
 Title: '.all()'
-Description: "Returns a new Promise that can be accessed as an array of resolved values of fulfulled Promises. It takes an iterable object, such as an Array, that contains one or more Promise objects. This is ideal when working with Promises that depend on one another's completion."
+Description: "The .all() method in JavaScript Promises helps us work with multiple tasks at once. It takes a group of tasks (Promises) and returns a new Promise.
+
+When we use .all(), all the tasks start together, and .all() waits for all of them to finish. It gives us a Promise that contains the results of all the tasks.
```
Comment (CBID2) [inline content/javascript/concepts/promise/terms/all/all.md:5]: [Proofreading/Nitpick] According to the [style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md#writing-entries), pronouns(e.g., I, we, us, our, they, you, he, she) can't be used. 
```suggestion
When using `.all()`, the tasks start together and wait for all of them to finish. `.all()` gives a Promise that contains the results of all the tasks.
```

File: content/javascript/concepts/promise/terms/all/all.md [hunk 2]
```diff
@@ -11,7 +15,11 @@ CatalogContent:
   - 'paths/full-stack-engineer-career-path'
 ---
 
-The `.all()` method returns a new Promise that can be accessed as an array of resolved values of fulfulled Promises. It takes an iterable object, such as an `Array`, that contains one or more Promise objects. This is ideal when working with Promises that depend on one another's completion.
+The .all() method in JavaScript Promises helps us work with multiple tasks at once. It takes a group of tasks (Promises) and returns a new Promise.
+
+When we use .all(), all the tasks start together, and .all() waits for all of them to finish. It gives us a Promise that contains the results of all the tasks.
```
Comment (CBID2) [inline content/javascript/concepts/promise/terms/all/all.md:20]: [Proofreading/Nitpick] These two lines share the same issue as line 5. Refer to [my feedback from there](https://github.com/Codecademy/docs/pull/2459/files#r1225974895) to help you.
```suggestion
The .all() method in JavaScript Promises helps us work with multiple tasks at once. It takes a group of tasks (Promises) and returns a new Promise.


When we use .all(), all the tasks start together, and .all() waits for all of them to finish. It gives us a Promise that contains the results of all the tasks.
```

File: content/javascript/concepts/promise/terms/all/all.md [hunk 3]
```diff
@@ -23,39 +31,61 @@ The `iterableObject` is usually an array of Promise objects. If the array is emp
...
+  .catch((error) => {
+    console.log('Error occurred:', error);
   });
+
 ```
+## Explanation
+
+In this example, we have three functions (fetchUserData(), fetchProductData(), and fetchOrderData()) that simulate API calls with a delay using setTimeout(). Each function returns a Promise that resolves with the respective data.
+
+We use Promise.all() to execute these promises in parallel and wait for all of them to resolve. The .then() callback receives an array of resolved values in the same order as the input promises. We destructure the array to extract the user data, product data, and order data, and log them to the console.
```
Comment (CBID2) [inline content/javascript/concepts/promise/terms/all/all.md:83]: [Proofreading/Nipick] Refer to [my comment on line 5](https://github.com/Codecademy/docs/pull/2459/files#r1225974895). It shares the same issue as this line.  
```suggestion
In this example, we have three functions (fetchUserData(), fetchProductData(), and fetchOrderData()) that simulate API calls with a delay using setTimeout(). Each function returns a Promise that resolves with the respective data.

We use Promise.all() to execute these promises in parallel and wait for all of them to resolve. The .then() callback receives an array of resolved values in the same order as the input promises. We destructure the array to extract the user data, product data, and order data, and log them to the console.
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi, @shivangnayar-dev! :) I gave some proofreading suggestions for your entry. I hope they help! :)

**Comment by shivangnayar-dev** [comment]
> Hi, @shivangnayar-dev! :) I gave some proofreading suggestions for your entry. I hope they help! :)

Thanks !!

**Comment by shivangnayar-dev** [comment]
> @shivangnayar-dev what the issue is asking for is to add new codebytes at the end of the two entries. Codebytes are examples of code that can be run by the reader of the entry. They should be different from the example given in the body of the entry, and should look something like this:
> 
> ```
> ## Codebyte Example
> 
> A sentence describing what the codebyte does:
> 
> ```codebyte/javascript
> the example code to run goes here
> ```
> ```
> 
> You can look at the codebytes for other JavaScript entries, like [here.](https://github.com/Codecademy/docs/edit/main/content/javascript/concepts/arrays/terms/indexOf/indexOf.md)
> 
> The issue you claimed wants codebytes to be added to both these entries:
> 
> * https://www.codecademy.com/resources/docs/javascript/promise/all
> * https://www.codecademy.com/resources/docs/javascript/promise/any

Ok i will do that!
Thanks

**Comment by SSwiniarski** [comment]
@shivangnayar-dev are you going to update this PR to add codebytes, or do you want to just submit it as-is?

**Comment by shivangnayar-dev** [comment]
> @shivangnayar-dev are you going to update this PR to add codebytes, or do you want to just submit it as-is?

Yes, i will add codebytes

**Comment by shivangnayar-dev** [comment]
> > @shivangnayar-dev are you going to update this PR to add codebytes, or do you want to just submit it as-is?
>

**Comment by shivangnayar-dev** [comment]
> > > @shivangnayar-dev are you going to update this PR to add codebytes, or do you want to just submit it as-is?



> @shivangnayar-dev are you going to update this PR to add codebytes, or do you want to just submit it as-is?

for now you can submit it as the codebyte has been created for .all() but for .any() is remaining

**Comment by shivangnayar-dev** [comment]
> @shivangnayar-dev are you going to update this PR to add codebytes, or do you want to just submit it as-is?

you can submit it as codebyte is created for .all()

**Comment by SSwiniarski** [comment]
> > @shivangnayar-dev are you going to update this PR to add codebytes, or do you want to just submit it as-is?
> 
> you can submit it as codebyte is created for .all()

@shivangnayar-dev in order to address the issue you also need to add a codebyte for .any() in this PR.

**Comment by SSwiniarski** [comment]
@shivangnayar-dev I am unassigning you from the original issue and reviewing this PR as-is without codebytes.

File: content/javascript/concepts/promise/terms/all/all.md [hunk 4]
```diff
@@ -1,6 +1,10 @@
 ---
 Title: '.all()'
-Description: "Returns a new Promise that can be accessed as an array of resolved values of fulfulled Promises. It takes an iterable object, such as an Array, that contains one or more Promise objects. This is ideal when working with Promises that depend on one another's completion."
+Description: "The .all() method in JavaScript Promises helps us work with multiple tasks at once. It takes a group of tasks (Promises) and returns a new Promise.
```
Comment (SSwiniarski) [inline content/javascript/concepts/promise/terms/all/all.md:3]:
```suggestion
Description: 'Takes a group of tasks (Promises) and returns a new Promise.'
```
Descriptions in the metadata are not multi-line.

File: content/javascript/concepts/promise/terms/all/all.md [hunk 5]
```diff
@@ -1,6 +1,10 @@
...
+
```
Comment (SSwiniarski) [inline content/javascript/concepts/promise/terms/all/all.md:4]:
```suggestion
```
Delete blank line.

File: content/javascript/concepts/promise/terms/all/all.md [hunk 6]
```diff
@@ -1,6 +1,10 @@
...
+When using `.all()`, the tasks start together and wait for all of them to finish. `.all()` gives a Promise that contains the results of all the tasks.
```
Comment (SSwiniarski) [inline content/javascript/concepts/promise/terms/all/all.md:5]:
```suggestion
```
Delete line

File: content/javascript/concepts/promise/terms/all/all.md [hunk 7]
```diff
@@ -11,7 +15,12 @@ CatalogContent:
   - 'paths/full-stack-engineer-career-path'
 ---
 
-The `.all()` method returns a new Promise that can be accessed as an array of resolved values of fulfulled Promises. It takes an iterable object, such as an `Array`, that contains one or more Promise objects. This is ideal when working with Promises that depend on one another's completion.
+The .all() method in JavaScript Promises helps us work with multiple tasks at once. It takes a group of tasks (Promises) and returns a new Promise.
```
Comment (SSwiniarski) [inline content/javascript/concepts/promise/terms/all/all.md:18]:
```suggestion
The **`.all()`** method in JavaScript Promises helps in working with multiple tasks at once. It takes a group of tasks (Promises) and returns a new Promise.
```
No first- or second- person pronouns.

File: content/javascript/concepts/promise/terms/all/all.md [hunk 8]
```diff
@@ -11,7 +15,12 @@ CatalogContent:
...
+
```
Comment (SSwiniarski) [inline content/javascript/concepts/promise/terms/all/all.md:19]:
```suggestion
```
Delete blank line

File: content/javascript/concepts/promise/terms/all/all.md [hunk 9]
```diff
@@ -11,7 +15,12 @@ CatalogContent:
...
+
+When we use .all(), all the tasks start together, and .all() waits for all of them to finish. It gives us a Promise that contains the results of all the tasks.
```
Comment (SSwiniarski) [inline content/javascript/concepts/promise/terms/all/all.md:21]:
```suggestion
When using `.all()`, all the tasks start together, and `.all()` waits for all of them to finish. It returns a Promise that contains the results of all the tasks.
```
No first- or second-person pronouns in Docs.

File: content/javascript/concepts/promise/terms/all/all.md [hunk 10]
```diff
@@ -23,39 +32,61 @@ The `iterableObject` is usually an array of Promise objects. If the array is emp
...
+    console.log('Order Data:', orderData);
   })
-  .finally(() => {
-    console.log('Operations for Promise.all() have finished.');
+  .catch((error) => {
+    console.log('Error occurred:', error);
   });
+
 ```
+## Explanation
```
Comment (SSwiniarski) [inline content/javascript/concepts/promise/terms/all/all.md:80]:
```suggestion
```
Remove the header.

File: content/javascript/concepts/promise/terms/all/all.md [hunk 11]
```diff
@@ -23,39 +32,61 @@ The `iterableObject` is usually an array of Promise objects. If the array is emp
...
+
```
Comment (SSwiniarski) [inline content/javascript/concepts/promise/terms/all/all.md:81]:
```suggestion
```
Delete blank line.

File: content/javascript/concepts/promise/terms/all/all.md [hunk 12]
```diff
@@ -23,39 +32,61 @@ The `iterableObject` is usually an array of Promise objects. If the array is emp
...
+In this example, we have three functions (fetchUserData(), fetchProductData(), and fetchOrderData()) that simulate API calls with a delay using setTimeout(). Each function returns a Promise that resolves with the respective data.
```
Comment (SSwiniarski) [inline content/javascript/concepts/promise/terms/all/all.md:82]:
```suggestion
In this example, there are three functions (`fetchUserData()`, `fetchProductData()`, and `fetchOrderData()`) that simulate API calls with a delay using `setTimeout()`. Each function returns a Promise that resolves with the respective data.
```
No first- or second- person pronouns in Docs.

File: content/javascript/concepts/promise/terms/all/all.md [hunk 13]
```diff
@@ -23,39 +32,61 @@ The `iterableObject` is usually an array of Promise objects. If the array is emp
...
+
+We use Promise.all() to execute these promises in parallel and wait for all of them to resolve. The .then() callback receives an array of resolved values in the same order as the input promises. We destructure the array to extract the user data, product data, and order data, and log them to the console.
```
Comment (SSwiniarski) [inline content/javascript/concepts/promise/terms/all/all.md:84]:
```suggestion
`Promise.all()` executes these promises in parallel and waits for all of them to resolve. The `.then()` callback receives an array of resolved values in the same order as the input promises. The example destructures the array to extract the user data, product data, and order data, and log them to the console.
```
No first- or second-person pronouns in Docs.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@shivangnayar-dev I reviewed your entry and had several comments.

**Review by SSwiniarski — APPROVED** [review]
@shivangnayar-dev In the interest of moving this along, I committed all my suggestions so it can go on to a second review.

**Comment by KTom101** [comment]
@SSwiniarski Taking a closer look at this PR. I noticed the Promise: [.all()](https://www.codecademy.com/resources/docs/javascript/promise/all) entry already has a codebyte. However, it looks like that update was made after this PR was made. So merging these changes will outdate the changes already made.

This can not be merged as is but if there are edits that we would like to include in the version we have now then I think this would need to be reopened. Let me know what you think on this.

cc @yangc95

**Comment by SSwiniarski** [comment]
@KTom101 do we just want to close this?

**Comment by CBID2** [comment]
> @KTom101 do we just want to close this?

I think it would be best

**Comment by KTom101** [comment]
I'm going ahead and closing this out since the original entry has been edited with the change mentioned in the issue. 

@shivangnayar-dev Feel free to work on another [issue](https://github.com/Codecademy/docs/issues?q=is%3Aissue+is%3Aopen+no%3Aassignee) but please make sure you understand what is being asked for first.

**Comment by KTom101** [comment]
I'm going ahead and closing this out since the original entry has been edited with the change mentioned in the issue.

@shivangnayar-dev Feel free to work on another [issue](https://github.com/Codecademy/docs/issues?q=is%3Aissue+is%3Aopen+no%3Aassignee) but please make sure you understand what is being asked for first.

**Pull request closed without merging**